### ISSUE TITLE
BetterDiscord installation fails

### ISSUE THREAD

**Issue by fhuewhd** [bb010g/betterdiscordctl#5]
I can install BetterDiscordCTL, and it seems to have installed, at least there it is in usr/local/bin, but when I put in "betterdiscordctl install -s /usr/share", I get this:
```
Updating asar...

/home/ab/.local/share/betterdiscordctl/asar/node_modules/asar/lib/asar.js:2
const fs = process.versions.electron ? require('original-fs') : require('fs')
^^^^^
SyntaxError: Use of const in strict mode.
    at Module._compile (module.js:439:25)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/home/ab/.local/share/betterdiscordctl/asar/node_modules/asar/bin/asar.js:2:12)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
sed: can't read /home/ab/.config/discord/0.0.4/modules/discord_desktop_core/core/app/mainScreen.js: No such file or directory
Cloning https://github.com/rauenzi/BetterDiscordApp...
ln: failed to create symbolic link ‘/home/ab/.config/discord/0.0.4/modules/discord_desktop_core/core/node_modules/betterdiscord’: No such file or directory
Installed. (Restart Discord if necessary.)
```
When I try to load Discord, it doesn't load, unless it's just updating really slowly - it's been hanging at the updater stage (the one that says "Readying the felines..." :-D ) for half an hour now.  Any advice, please?

**Comment by ObserverOfTime** [comment]
This seems to be a Node issue. Try [this](https://stackoverflow.com/a/35218558/9027302).

**Comment by bb010g** [comment]
What distribution are you using, and how did you install Node?

**Comment by bb010g** [comment]
Also, we may want to look into more checks for failure, as painful as bash may make them. :/

**Comment by ObserverOfTime** [comment]
We could add individual checks after the node-related commands that seem to fail a lot, or do something with [ERR or DEBUG](https://stackoverflow.com/questions/42218009/how-to-tell-if-any-command-in-bash-script-failed-non-zero-exit-status).

**Comment by fhuewhd** [comment]
Issue fixed by uninstalling everything very thoroughly and then reinstalling again using BetterDiscordCTL method, following exact instructions word for word. I'd been trying the main method and then the BetterDiscordCTL method haphazardly, switching to one when the other failed, and I think that must have muddled things up. Lesson, don't mix the two, if one fails then uzninstall BetterDiscrod, Node and NPM completely before trying again with the other. 
You were quite right, it was NodeJS and I believe I can see why now.  To install NodeJS, I typed "sudo apt install nodejs-legacy", as per the instructions in the Tutorial page on the BetterDiscord wiki.  I thought at the time it was odd that it specified a "legacy" version, but that seemed to be what it wanted.  Of course, that was the OTHER installation route - by the time I got to trying BetterDisordCTL again, I was fairly confused and didn't remember about the "legacy" thing, I just remembered that I'd already installed something called NodeJS and assumed that that was what it wanted.  I did as the StackExchange instructions said and then removed BetterDiscordCTL with Synaptic Package Manger and tried again from scratch, and this time it installed successfully.  
Somewhere along the way I also seem to have UN-broken Discord.  Thanks very much for all the help.  (Do yyou want me to close the issue or are you two still talking?)
